### PR TITLE
feat(e2e-setup): Update script to setup required assets for AAD auth tests

### PR DESCRIPTION
**Pull request by vinagesh**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

### CODE REVIEW

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1 [hunk 1]
```diff
@@ -339,25 +340,6 @@ if ($InstallDependencies)
     az extension add --name azure-iot
 }
 
-#################################################################################################################################################
```
Comment (vinagesh) [inline e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1:342]: moved below and updated to using the new az command to create service principal

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1 [hunk 2]
```diff
@@ -436,6 +418,26 @@ $keyVaultName = az deployment group show -g $ResourceGroup -n $deploymentName --
...
+# Configure an AAD app and create self signed certs and get the bytes to generate more content info.
+#################################################################################################################################################
+Write-Host "`nCreating App Registration $logAnalyticsAppRegnName"
+$logAnalyticsAppRegUrl = "http://$logAnalyticsAppRegnName"
+az ad sp create-for-rbac -n $logAnalyticsAppRegUrl --role "Reader" --scope $resourceGroupId --output none
+$logAnalyticsAppId = az ad app list --display-name $logAnalyticsAppRegnName --query "[?displayName=='$logAnalyticsAppRegnName'].appId" --output tsv
+Write-Host "`nApplication $logAnalyticsAppRegnName with Id $logAnalyticsAppId was created successfully."
+
+#################################################################################################################################################
+# Configure an AAD app to perform IoT hub data actions.
```
Comment (vinagesh) [inline e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1:431]: Newly added for aad auth tests. az ad sp create-for-rbac ensures we don't need any retry logic.

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1 [hunk 3]
```diff
@@ -554,36 +556,6 @@ az iot dps enrollment create `
     --certificate-path $individualDeviceCertPath `
     --output none
 
-# The Service Principal takes a while to get propogated and if a different endpoint is hit before that, trying to grant a permission will fail.
-# Adding retries so that we can grant the permissions successfully without re-running the script.
-Write-Host "`nGranting $logAnalyticsAppId Reader role assignment to the $ResourceGroup resource group."
```
Comment (vinagesh) [inline e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1:559]: az ad sp create-for-rbac ensures we don't need any retry logic. So removing this logic now.

**Review by timtay-microsoft — APPROVED** [review]

**Review by drwill-ms — APPROVED** [review]

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1 [hunk 4]
```diff
@@ -436,6 +418,26 @@ $keyVaultName = az deployment group show -g $ResourceGroup -n $deploymentName --
...
+Write-Host "`nCreating App Registration $logAnalyticsAppRegnName"
+$logAnalyticsAppRegUrl = "http://$logAnalyticsAppRegnName"
+az ad sp create-for-rbac -n $logAnalyticsAppRegUrl --role "Reader" --scope $resourceGroupId --output none
+$logAnalyticsAppId = az ad app list --display-name $logAnalyticsAppRegnName --query "[?displayName=='$logAnalyticsAppRegnName'].appId" --output tsv
+Write-Host "`nApplication $logAnalyticsAppRegnName with Id $logAnalyticsAppId was created successfully."
+
+#################################################################################################################################################
+# Configure an AAD app to perform IoT hub data actions.
+#################################################################################################################################################
+Write-Host "`nCreating App Registration $iotHubAadTestAppRegName"
```
Comment (abhipsaMisra) [inline e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1:433]: q - could we have reused the existing aad app for iot hub tests as well?

Comment (vinagesh) [reply e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1:433]: I could not do two different role assignments through the PowerShell command. Maybe there is a way to do it but I did not want to spend too much time on it. Technically it is possible if we did it manually on the portal.

**Pull request merged**